### PR TITLE
Add support for dropping files from other browser windows

### PR DESCRIPTION
**Pull request by mtlewis**
* Add util for retrieving image url from data transfer items.
* Use it to attempt to retrieve the image src from the drop event.

**Note**: this will only work in IE10+ and evergreen browsers - in older browsers the DOMParser either doesn't exist or doesn't support html parsing.  The obvious alternative (loading the html in the dataTransferItem in a detached div) is a bad idea, as it opens up the possibility of XSS (e.g. via the onload attribute of an img).

**Edit**: I realised that the current build system didn't support bundling all the sources into a single file, so I've added rollup for building the component bundle.  Also added Jest and wrote some unit tests for the modules - hopefully we can use this as a base for splitting up and testing the rest of the component.

### CODE REVIEW

**Comment by mtlewis** [comment]
Not sure why the build is failing here - happy to help debug, but as I can't see the logs from CI and it works locally I think I'll need some help tracking down the cause.

**Comment by dan-lee** [comment]
Cool changes 👍 awesome that you added some tests!
We will look at it more closely soon

Here are the CI logs:
> 1:25:27 PM: $ jest
1:25:30 PM: PASS src/utils/retrieve-image-url.test.js
1:25:30 PM: FAIL src/utils/parse-dom.test.js
● parseDOM › No DOMParser available › retuns null
>
> expect(received).toBeNull()
>
> Expected value to be null, instead received
{"location": null, Symbol(impl): {"_URL": [Object], "_activeNodeIterators": Array [], "_activeNodeIteratorsMax": 10, "_agentOptions": undefined, "_attached": true, "_childNodesList": null, "_childrenList": null, "_contentType": "text/html", "_cookieJar": [Object], "_core": [Object], "_currentScript": null, "_customResourceLoader": undefined, "_defaultView": null, "_documentElement": null, "_encoding": "UTF-8", "_eventListeners": [Object], "_global": undefined, "_history": [Object], "_htmlToDom": [Object], "_ids": [Object], "_implementation": [Object], "_lastFocusedElement": null, "_lastModified": "07/16/2017 11:25:30", "_location": [Object], "_memoizedQueries": [Object], "_origin": "null", "_ownerDocument": [Circular], "_parsingMode": "html", "_pool": undefined, "_proxy": undefined, "_queue": [Object], "_referrer": "", "_requestManager": [Object], "_strictSSL": undefined, "_version": 2, "nodeType": 9, "readyState": "complete", Symbol(DOM SymbolTree): [Object], Symbol(wrapper): [Circular]}}
>
> at Object.<anonymous> (src/utils/parse-dom.test.js:51:22)
at Promise.resolve.then.el (node_modules/p-map/index.js:42:16)
at process._tickCallback (internal/process/next_tick.js:109:7)
1:25:30 PM: Test Suites: 1 failed, 1 passed, 2 total
Tests: 1 failed, 10 passed, 11 total

**Comment by mtlewis** [comment]
That's weird - works on my machine ;)

Will take a look shortly and see if I can see why they might be failing.

**Comment by mtlewis** [comment]
D'oh - it looks like I actually broke this test when I reorganised the setup and teardown.  You can't use the `delete` keyword on the global object - switching to setting it to undefined fixes this :)

**Comment by dan-lee** [comment]
You could merge master into here to get the `.travis.yml`, so we have actual CI.

File: package.json [hunk 1]
```diff
@@ -38,7 +40,9 @@
     "image"
   ],
   "dependencies": {
-    "prop-types": "^15.5.8"
+    "prop-types": "^15.5.8",
+    "react": "^15.5.4",
```
Comment (dan-lee) [inline package.json]: I think we should leave them as `peerDependencies` + `devDependencies`, or why did you change that?

Comment (mtlewis) [inline package.json]: Ah - I missed that they were in peerDependencies :).  Happy to remove them.  Should we move prop-types there too?

**Comment by mtlewis** [comment]
Rebased on latest master - working now :)

File: package.json (hunk 1 above)
Comment (dan-lee) [inline package.json]: No, I think we should keep that, we cannot rely on that being provided by the using app.

Comment (mtlewis) [inline package.json]: As long as it's in one or the other, no sense bikeshedding about it.  Sorry, if I'd seen the entries in devDependencies I never would have made this change, don't want to get into a bikeshed conversation ;)

Comment (mtlewis) [inline package.json]: Done :)

**Comment by dan-lee** [comment]
Looks great to me! We even shave off 5kb with rollup 👍 
I think providing the source map is also a plus.

@mosch what do you think?

**Comment by mosch** [comment]
Really cool work @mtlewis - you really help making this better!

**Comment by mtlewis** [comment]
Thanks @mosch :)
